### PR TITLE
Client robustnes against temporary glitches

### PR DESCRIPTION
**Pull request by ChristianMayer**

### CODE REVIEW

File: client/source/class/cv/io/Client.js [hunk 1]
```diff
@@ -491,7 +492,8 @@ qx.Class.define('cv.io.Client', {
           }
           if (options.listeners) {
             Object.getOwnPropertyNames(options.listeners).forEach(function(eventName) {
-              ajaxRequest.addListener(eventName, options.listeners[eventName], context);
+              var qxEventName = 'error' !== eventName ? eventName : 'statusError';
+              ajaxRequest.addListener(qxEventName, options.listeners[eventName], context);
```
Comment (peuter) [inline client/source/class/cv/io/Client.js]: The 'statusError' listener is always added to ajaxRequest (see some lines below this one). What is the reason for this change? IMHO this change would result in two listeners to the 'statusError' event and no listener to the 'error' event.

File: client/source/class/cv/io/transport/LongPolling.js [hunk 2]
```diff
@@ -208,6 +211,14 @@ qx.Class.define('cv.io.transport.LongPolling', {
...
       "qx": function (ev) {
         var req = ev.getTarget();
+        // check for temporary server errors and retry a few times
+        if([408,444,499,502,503,504].indexOf(req.getStatus()) >= 0 && this.retryServerErrorCounter < this.client.backend.maxRetries) {
+          this.info('Temporary connection problem (status: ' + req.getStatus() + ') - retry count: ' + this.retryServerErrorCounter);
+          this.retryServerErrorCounter++;
+          req.serverErrorHandled = true;
+          this.restart();
+          return;
+        }
```
Comment (peuter) [inline client/source/class/cv/io/transport/LongPolling.js]: The retry-handling feature should also be added to the jquery version of the client. Otherwise the two client releases would behave differently.

File: config.json [hunk 3]
```diff
@@ -296,6 +296,7 @@
       {
         "cv.build": "source",
         "cv.sentry": false,
+        "qx.dynlocale": false,
```
Comment (peuter) [inline config.json]: Deactivating the dynamic locale switching is fine, as we do not need it in the client. But out of curiosity: Did you experience any problems with this feature here, that made you change the setting?

File: client/source/class/cv/io/Client.js (hunk 1 above)
Comment (ChristianMayer) [reply client/source/class/cv/io/Client.js]: That's the other way round:
This line says: "please use `eventName` - except when it's `'error'`, then replace that with `'statusError'`"

There can be many listeners to `statusError`, one from the Client itself (line 507) and one per `options.listeners` at this line.

The reason for that change was that the listener for the `error` wasn't called, even when I made the server to give back a few 502 status codes

File: config.json (hunk 3 above)
Comment (ChristianMayer) [reply config.json]: Yes, there were some problems (cf. Gitter conversation).
The code is expecting a String in some places but only got the translation-object instead.

File: client/source/class/cv/io/transport/LongPolling.js (hunk 2 above)
Comment (ChristianMayer) [reply client/source/class/cv/io/transport/LongPolling.js]: As I have no way to test as the moment I didn't want to take the risk for such a change at a very critical key component of the CV.
The PR is "just" improving robustness, so I think it's ok to have a slightly different behaviour

Comment (peuter) [reply client/source/class/cv/io/transport/LongPolling.js]: Ok, I assume that there are not many (or even no) users for the standalone client in the jquery version, I guess we can live with that difference.
